### PR TITLE
Network interface s/netmask6/prefix6/

### PR DESCRIPTION
**Pull request by isimluk**
With IPv6, the literature prefers to talk about `prefix length`. The netmask is still technically correct, but only rarely used.

We don't expect our users to read&type stuff like: `"ffff:ffff:ffff:ffff:ffff:ffff::"` we want them to say just `96` or `42`.

### CODE REVIEW

**Comment by isimluk** [comment]
Nothing under ManageIQ ☂️  uses `netmask6`, so we can safely replace.

**Comment by isimluk** [comment]
> I think that if netmask is still technically correct, I would rather have the interface be consistent between 
> IPv4 and IPv6 than have the naming be slightly more correct.

Nod. Although, I believe this goes deeper.

Let's take a look what others do.
 * The output of `ip` command uses prefixes exclusively for both ipv4 and ipv6.
 * The `nmcli` tool seems to use prefixes exclusivelly.
 * The syntax of NetworkManager config files requires netmask for ipv4 and prefixes in ipv6. The ``
 * The UI tools (like cockpit or NetworkManager-gnome) use netmaks for ipv4 and prefix length for ipv6.

Note that NM config, ip command and nmcli command take `*/prefix` as a essential part of ipaddress. There are no specific way to say `netmask` they just say:  IPADDR=`::1/127`. On modern Linux, the only low-level place that takes netmask seems to be `NETMASK` in NM config.

My theory is that people are used to netmask for IPv4 addresses because they remember the first time they connected their device to a network. They remember when CIDR was the new trick they learned. However, they never used to see netmask6.

So, I am inclined to allow for the slight API inconsistency. What about rewriting IPv4 API to use prefix as well?

**Comment by isimluk** [comment]
> Also, even if we were to go this route, it would be good to preserve the yardoc and have a separate
> commit for the refactoring rather than a commit for removing and then re-adding a functionally
> equivalent method.

Wrote the docs and squashed the commits.

The motivation for having the commits separated was that it is not refactoring, rather removal of unused method and addition of a method that will be useful. It is not clear for outside contributor if this project has aspiration to become a library or not. If the aspiration is there, then we should keep the interface intact. And I would need to drop the first commit. :)

**Comment by jrafanie** [comment]
> I think that if netmask is still technically correct, I would rather have the interface be consistent between
> IPv4 and IPv6 than have the naming be slightly more correct.
> Nod. Although, I believe this goes deeper.
> 
> Let's take a look what others do.
> 
> The output of ip command uses prefixes exclusively for both ipv4 and ipv6.
> The nmcli tool seems to use prefixes exclusivelly.
> The syntax of NetworkManager config files requires netmask for ipv4 and prefixes in ipv6. The ``
> The UI tools (like cockpit or NetworkManager-gnome) use netmaks for ipv4 and prefix length for ipv6.
> Note that NM config, ip command and nmcli command take */prefix as a essential part of ipaddress. There are no specific way to say netmask they just say: IPADDR=::1/127. On modern Linux, the only low-level place that takes netmask seems to be NETMASK in NM config.
> 
> My theory is that people are used to netmask for IPv4 addresses because they remember the first time they connected their device to a network. They remember when CIDR was the new trick they learned. However, they never used to see netmask6.
> 
> So, I am inclined to allow for the slight API inconsistency. What about rewriting IPv4 API to use prefix as well?

@isimluk @carbonin I agree with both of you.  We should be consistent between both IPv4 and v6 API but clearly the tools aren't all consistent.  Even though we're probably not using the current netmask6, technically renaming this is a breaking change and we should deprecate it before removing it.

I'm wondering if we can solve all of this by aliasing them so we have both prefix/prefix6 and netmask/netmask6.  The implementation can use `prefix/prefix6`  with the netmasks aliasing prefix.  Our client code can then choose to use either one.   `prefix` if that's the more widely used, or `netmask` if the client code is in an area that consistently uses `netmask`.

**Comment by jrafanie** [comment]
> Wrote the docs and squashed the commits.
> 
> The motivation for having the commits separated was that it is not refactoring, rather removal of unused method and addition of a method that will be useful. It is not clear for outside contributor if this project has aspiration to become a library or not. If the aspiration is there, then we should keep the interface intact. And I would need to drop the first commit. :)

@isimluk to be fair, this is a trivial change so it's pretty obvious what's going on.  But, it's a good habit to make it easy for others see exactly what was done.  If the remove and add with a different name are separate commits and in a much larger pull request, it would be hard to follow what was going on.  By doing the rename in one commit and any refactorings in another, me, others, future you, can easily walk the commits one by one and know exactly what happened.  It's always possible to squash commits later if they don't need to be so small but to be honest, I can't recall ever disliking small "singular functional change" commits.  Squashing typos, fixes for prior commits, or the same functional change split among several commits is perfectly fine.

**Comment by carbonin** [comment]
>I'm wondering if we can solve all of this by aliasing them so we have both prefix/prefix6 and netmask/netmask6. The implementation can use prefix/prefix6 with the netmasks aliasing prefix. Our client code can then choose to use either one. prefix if that's the more widely used, or netmask if the client code is in an area that consistently uses netmask.

I like this idea, but I'm not sure an alias will do the job. Netmask is a full address where prefix is just a number so we can't possibly have one method that returns both.

Maybe it would be better to have separate methods that interpret the same value on the object in different ways? We can keep the backend hash value as a prefix then when someone calls `netmask6` we return that prefix as an IPv6 netmask. This way we add functionality and don't break the API.

**Comment by Fryguy** [comment]
:+1: for aliasing.  In particular, removing netmask6 requires a major version bump in the gem.  I don't necessarily care about the version number, but we would also have to release a minor version with a deprecation warning.

**Comment by isimluk** [comment]
Ok, I'll start over.
